### ISSUE TITLE
fails dramatically 

### ISSUE THREAD

**Issue by typemytype** [typemytype/booleanOperations#48]
```xml
<?xml version="1.0" encoding="UTF-8"?>
<glyph name="A" format="2">
  <advance width="500"/>
  <outline>
    <contour>
      <point x="100" y="700" type="line"/>
      <point x="500" y="700" type="line"/>
      <point x="100" y="900" type="line"/>
      <point x="500" y="500" type="line"/>
    </contour>
  </outline>
</glyph>
```

the contour is gone after a boolean operation

**Comment by typemytype** [comment]
the [area](https://github.com/typemytype/booleanOperations/blob/master/Lib/booleanOperations/booleanOperationManager.py#L39)  is zero here in this case...

**Comment by anthrotype** [comment]
ouch

**Comment by anthrotype** [comment]
@typemytype I forgot why did we do that?

**Comment by typemytype** [comment]
real zero area paths cause errors later on, did test yet with such a path.

**Comment by anthrotype** [comment]
by real zero area paths, perhaps you mean a closed path with all the points on top of each other?

**Comment by justvanrossum** [comment]
> by real zero area paths, perhaps you mean a closed path with all the points on top of each other?

I think rather situations like:

```python
pen.moveTo((100, 100))
pen.lineTo((300, 300))
pen.closePath()
```

**Comment by typemytype** [comment]
real zero aera paths are paths with no visible fill, could be all points on top, could be like Justs example.

the case above should return something ;) and definitely not ignore the contour

**Comment by typemytype** [comment]
some cases and a narrowed down fix below, used in the PR

```xml
<?xml version="1.0" encoding="UTF-8"?>
<glyph name="A" format="2">
  <advance width="500"/>
  <outline>
    <contour>
      <point x="100" y="700" type="line"/>
      <point x="500" y="700" type="line"/>
      <point x="100" y="920" type="line"/>
      <point x="500" y="500" type="line"/>
    </contour>
    <contour>
      <point x="299" y="1003" type="move"/>
      <point x="591" y="1030" type="line"/>
    </contour>
    <contour>
      <point x="460" y="943" type="line"/>
      <point x="605" y="958" type="line"/>
    </contour>
    <contour>
      <point x="103.0" y="1018" type="line"/>
      <point x="103.0" y="1017.5" type="line"/>
    </contour>
    <contour>
      <point x="103.0" y="1017.5" type="line"/>
      <point x="103.0" y="1017.5" type="line"/>
      <point x="103.0" y="1017.5" type="line"/>
      <point x="103.0" y="1017.5" type="line"/>
      <point x="103.0" y="1018.0" type="line"/>
    </contour>
  </outline>
</glyph>
```

```python

import pyclipper

glyph = CurrentGlyph()

subjectContours = []
for contour in glyph:
    subjectContours.append([(p.x, p.y) for p in contour.points])
    r = pyclipper.Area(subjectContours[-1])
    print(r)


pc = pyclipper.Pyclipper()
for subjectContour in subjectContours:
    try:
        pc.AddPath(subjectContour, pyclipper.PT_SUBJECT)
    except pyclipper.ClipperException:
        pass

r = pc.Execute(pyclipper.CT_UNION, pyclipper.PFT_NONZERO, pyclipper.PFT_NONZERO)
 
print(r)
```

**Comment by anthrotype** [comment]
Fixed by #51
